metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_queries.py

```diff
@@ -568,8 +629,12 @@ def _build_enriched_query_log_query(
         q.role_name AS "ROLE_NAME",
         a.direct_objects_accessed,
         a.objects_modified,
+        a.object_modified_by_ddl
     FROM deduplicated_queries q
     JOIN filtered_access_history a USING (query_id)
+    -- This qualify ignores more than 1 rows from same query, such as for ddl swap
+    QUALIFY
+        ROW_NUMBER() OVER (PARTITION BY query_id ORDER BY start_time DESC) = 1
```

Comment:
we have the deduplicated_queries cte - should we push stuff down to that?

Reply:
As we are now ignoring duplicate swaps, we can in fact remove this qualify as it is fine to process same swap query multiple times.